### ISSUE TITLE
[FR]: Snapshots Improvements

### ISSUE THREAD

**Issue by jtrv** [mbnuqw/sidebery#1815]
### Description

I think the snapshots feature could be incredibly useful with a few additions and a more ergonomic UI/UX.

[Tab stash's sidebar](https://addons.mozilla.org/en-US/firefox/addon/tab-stash/) is a great example of how we can improve the snapshot UX: 

- Improve upon the existing feature with additional functionality:
	- Add tabs to stash
	- Remove tabs from stash
	- Save partial stash
- Replace "snapshot" terminology with the more semantic/flexible "stash" aligning with new functionality
- Improve UI / UX and align it with other sidebar features e.g. bookmarks, history
- Stash tree, pins, etc, as 1:1 image of what appears in the tabs sidebar
- Save stash in a nameable folder with the date/time as default name

**Comment by segln** [comment]
Agreed. The snapshot feature is well made but currently it is hard to use for anything other than restoring tabs.

It would be great if it could function like disk storage for arbitrary tab structures as well.
